homeassistant/components/vesync/switch.py

```diff
@@ -45,53 +71,45 @@ def _setup_entities(
     async_add_entities,
     coordinator: VeSyncDataCoordinator,
 ):
-    """Check if device is a switch and add entity."""
-    entities: list[VeSyncBaseSwitch] = []
-    for dev in devices:
-        if DEV_TYPE_TO_HA.get(dev.device_type) == "outlet":
-            entities.append(VeSyncSwitchHA(dev, coordinator))
-        elif DEV_TYPE_TO_HA.get(dev.device_type) == "switch":
-            entities.append(VeSyncLightSwitch(dev, coordinator))
-
-    async_add_entities(entities, update_before_add=True)
+    """Check if device is online and add entity."""
+    async_add_entities(
+        VeSyncSwitchEntity(dev, description, coordinator)
+        for dev in devices
+        for description in SENSOR_DESCRIPTIONS
+        if description.exists_fn(dev)
+    )
 
 
-class VeSyncBaseSwitch(VeSyncBaseEntity, SwitchEntity):
-    """Base class for VeSync switch Device Representations."""
+class VeSyncSwitchEntity(SwitchEntity, VeSyncBaseEntity):
+    """VeSync switch entity class."""
 
-    _attr_name = None
+    entity_description: VeSyncSwitchEntityDescription
 
-    def turn_on(self, **kwargs: Any) -> None:
-        """Turn the device on."""
-        self.device.turn_on()
+    def __init__(
```

Comment:
We need to define `_attr_unique_id` otherwise adding another switch entity fails.

Something like this will work - `self._attr_unique_id = f"{super().unique_id}-{description.key}"`

Reply:
For the two devices exposed today if we change this it will be a breaking change.    However I could add a filter to only do the above unique ID on ones other than the outlet and switch on we offer today.  It would set us up for future. 

Thoughts on that approach?

Reply:
I suppose that will work. @joostlek might have reviewed code which ran into similar shortcoming and if the _attr_unique_id was defined as a breaking change.

Reply:
Oh that is a good catch. If that is the case we should migrate first. I don't feel like merging this as it will create entities with not unique unique ids. So we should have a preliminary PR where we migrate all the unique ids of the switches to something else and then we can set up new ones

Reply:
Okay,  today this code doesn't conflict or impact unique ID, however since trying to setup for growth makes sense we fix this now.   I will update this code to have the unique ID as mentioned above and chat with iprak around getting a PR for that migration that will merge first.